ztp/siteconfig-generator/siteConfig/siteConfig.go

```diff
@@ -228,6 +231,57 @@ type TangConfig struct {
 	Thumbprint string `yaml:"thumbprint" json:"thp"`
 }
 
+type DiskPartition struct {
+	Device     string       `yaml:"device"`
+	Partitions []Partitions `yaml:"partitions"`
+}
+
+type Partitions struct {
+	MountPoint    string `yaml:"mount_point" `
+	Size          int    `yaml:"size"`
+	Start         int    `yaml:"start"`
+	MountFileName string `yaml:"-"`
+	Label         string `yaml:"-"`
+}
+
+type PartitionsUserInput Partitions
+
+func (prt *Partitions) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	var userInput = PartitionsUserInput{}
+
+	err := unmarshal(&userInput)
+
+	if userInput.Start < 25000 {
+		// 25000 min value, otherwise root partition is too small and the installation will fail
+		err = fmt.Errorf("start value too small. must be over 25000")
```

Comment:
these `err` statements get over written if multiple user inputs are wrong, would be best to concat them, so we don't shadow errors presented to the user

Reply:
Now all the errors are concatenated into an array and returned after a join.